backend/inspirehep/records/marshmallow/literature/ui.py

```diff
@@ -185,18 +188,26 @@ def get_number_of_references(self, data):
         references = data.get("references")
         return self.get_len_or_missing(references)
 
-    def get_linked_book(self, data):
-        parent = get_parent_record(data)
-        if parent and "titles" in parent and "control_number" in parent:
-            endpoint = PidStoreBase.get_endpoint_from_pid_type(
-                PidStoreBase.get_pid_type_from_schema(data["$schema"])
-            )
-            endpoint_item = f"invenio_records_rest.{endpoint}_item"
-            ref = get_value(parent, "self.$ref") or url_for(
-                endpoint_item, pid_value=parent["control_number"], _external=True
+    def get_linked_books(self, data):
+        parents = get_parent_records(data)
+        pages = get_pages(data)
+
+        records = [
+            {
+                **parent["titles"][0],
+                "record": get_record_ref(parent["control_number"], "literature"),
+            }
+            for parent in parents
+            if parent and "titles" in parent and "control_number" in parent
+        ]
+        merged_list = [
+            {"page_start": start, "page_end": end, **record}
+            for start, end, record in zip(
+                pages["page_start"], pages["page_end"], records
             )
-            return {**parent["titles"][0], "record": {"$ref": ref}}
-        return None
+        ]
+
+        return merged_list
```

Comment:
Sorry for that,  but it's better to return None if empty, hence 

```
return merged_list or None 
```

should be enough